<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from robo.urdf.xacro                | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="robogardener">
  <!-- Base Link -->
  <link name="base_link">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.3 0.2 0.1"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <box size="0.3 0.2 0.1"/>
      </geometry>
      <material name="white"/>
    </visual>
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="6"/>
      <inertia ixx="0.025000000000000005" ixy="0.0" ixz="0.0" iyy="0.05" iyz="0.0" izz="0.065"/>
    </inertial>
  </link>
  <!--xacro:pantograph_arm/ -->
  <joint name="left_f_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="left_f_wheel"/>
    <origin rpy="0 0 0" xyz="0.075 -0.1 0"/>
    <axis xyz="0 1 0"/>
    <dynamics damping="0.2"/>
    <limit effort="100" velocity="1.0"/>
  </joint>
  <!-- left wheel Link -->
  <link name="left_f_wheel">
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="8.333333333333334e-05" ixy="0.0" ixz="0.0" iyy="8.333333333333334e-05" iyz="0.0" izz="0.00012500000000000003"/>
    </inertial>
  </link>
  <joint name="left_b_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="left_b_wheel"/>
    <origin rpy="0 0 0" xyz="-0.075 -0.1 0"/>
    <axis xyz="0.707 0.707 0"/>
    <dynamics damping="0.2"/>
    <limit effort="100" velocity="1.0"/>
  </joint>
  <!-- left wheel Link -->
  <link name="left_b_wheel">
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="8.333333333333334e-05" ixy="0.0" ixz="0.0" iyy="8.333333333333334e-05" iyz="0.0" izz="0.00012500000000000003"/>
    </inertial>
  </link>
  <joint name="right_f_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="right_f_wheel"/>
    <origin rpy="0 0 0" xyz="0.075 0.1 0"/>
    <axis xyz="0.707 0.707 0"/>
    <dynamics damping="0.2"/>
    <limit effort="100" velocity="1.0"/>
  </joint>
  <!-- right wheel Link -->
  <link name="right_f_wheel">
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="8.333333333333334e-05" ixy="0.0" ixz="0.0" iyy="8.333333333333334e-05" iyz="0.0" izz="0.00012500000000000003"/>
    </inertial>
  </link>
  <joint name="right_b_wheel_joint" type="continuous">
    <parent link="base_link"/>
    <child link="right_b_wheel"/>
    <origin rpy="0 0 0" xyz="-0.075 0.1 0"/>
    <axis xyz="0.707 0.707 0"/>
    <dynamics damping="0.2"/>
    <limit effort="100" velocity="1.0"/>
  </joint>
  <!-- right wheel Link -->
  <link name="right_b_wheel">
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.05" radius="0.05"/>
      </geometry>
      <material name="yellow"/>
    </visual>
    <inertial>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <mass value="0.1"/>
      <inertia ixx="8.333333333333334e-05" ixy="0.0" ixz="0.0" iyy="8.333333333333334e-05" iyz="0.0" izz="0.00012500000000000003"/>
    </inertial>
  </link>
  <!-- Material Color -->
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="yellow">
    <color rgba="0.8 0.8 0.0 1.0"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="green">
    <color rgba="0.0 1.0 0.0 1.0"/>
  </material>
  <ros2_control name="robogardener" type="system">
    <hardware>
      <plugin>gz_ros2_control/GazeboSystem</plugin>
      <param name="robot_description">{load_urdf('/home/noi/pw2/install/robogardener/share/robogardener/urdf/robo.urdf.xacro')}</param>
    </hardware>
    <joint name="left_f_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="left_b_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="right_f_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="right_b_wheel_joint">
      <command_interface name="velocity"/>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
</robot>
